src/systempolicy/systemPolicy.go

```diff
@@ -788,6 +789,42 @@ func mergeSysPolicies(pols []types.KnoxSystemPolicy) []types.KnoxSystemPolicy {
 			})
 		}
 	}
+	// If a binary is allowed without fromSource, Convert the file access for it to global access
+
+	// Pre-compute: needed to optimize search
+
+	gloablBinaries := []string{}
+
+	for _, pol := range results {
+		for _, processPathRule := range pol.Spec.Process.MatchPaths {
+			if len(processPathRule.FromSource) == 0 {
+				gloablBinaries = append(gloablBinaries, processPathRule.Path)
+			}
+		}
+		for _, processDirRule := range pol.Spec.Process.MatchDirectories {
+			if len(processDirRule.FromSource) == 0 {
+				gloablBinaries = append(gloablBinaries, processDirRule.Dir)
+			}
+		}
+	}
+
+	for _, pol := range results {
+		for i, filePath := range pol.Spec.File.MatchPaths {
+			for _, binary := range filePath.FromSource {
+				if slices.Contains(gloablBinaries, binary.Path) {
```

Comment:
It should be something like strings.HasPrefix() for each path right. Are we not comparing to full fromSource path ?

Reply:
We dont need that because fromsource has absolute binary paths.